### PR TITLE
Implemented library uri support for FlutterFragments and FlutterActivities

### PR DESCRIPTION
**Pull request by gaaclarke**
issue https://github.com/flutter/flutter/issues/91841

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java [hunk 1]
```diff
@@ -824,6 +825,32 @@ public String getDartEntrypointFunctionName() {
     }
   }
 
+  /**
+   * The Dart library uri for the entrypoint that will be executed as soon as the Dart snapshot is
+   * loaded.
+   *
+   * <p>Example value: "package:foo/bar.dart"
+   *
+   * <p>This preference can be controlled by setting a {@code <meta-data>} called {@link
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:834]: btw, the manifest can have dynamic values that are passed from the tool to Gradle, then copied into the final manifest. e.g. https://github.com/flutter/flutter/blob/master/packages/flutter_tools/templates/app_shared/android.tmpl/app/src/main/AndroidManifest.xml.tmpl#L5

cc @GaryQian

Comment (blasten) [reply shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:834]: This needs another migration of existing manifest files

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java [hunk 2]
```diff
@@ -824,6 +825,32 @@ public String getDartEntrypointFunctionName() {
     }
   }
 
+  /**
+   * The Dart library uri for the entrypoint that will be executed as soon as the Dart snapshot is
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterActivity.java:829]: nit: uppercase for URI

File: shell/platform/android/io/flutter/embedding/android/FlutterFragment.java [hunk 3]
```diff
@@ -105,6 +105,8 @@
 
   /** The Dart entrypoint method name that is executed upon initialization. */
   protected static final String ARG_DART_ENTRYPOINT = "dart_entrypoint";
+  /** */
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterFragment.java:108]: nit: missing comment

File: shell/platform/android/io/flutter/embedding/android/FlutterFragment.java [hunk 4]
```diff
@@ -1026,6 +1036,20 @@ public String getDartEntrypointFunctionName() {
     return getArguments().getString(ARG_DART_ENTRYPOINT, "main");
   }
 
+  /**
+   * Returns the library uri of the Dart method that this {@code FlutterFragment} should execute to
```
Comment (blasten) [inline shell/platform/android/io/flutter/embedding/android/FlutterFragment.java:1040]: nit: uppercase URI

**Review by blasten — APPROVED** [review]
lgtm

**Comment by ColdPaleLight** [comment]
I guess that we also need to add `getDartEntrypointLibraryUri ` to `FlutterFragmentActivity`, just like `getDartEntrypointFunctionName`.

File: shell/platform/android/io/flutter/embedding/android/FlutterFragment.java (hunk 4 above)
Comment (gaaclarke) [reply shell/platform/android/io/flutter/embedding/android/FlutterFragment.java]: done

File: shell/platform/android/io/flutter/embedding/android/FlutterFragment.java (hunk 3 above)
Comment (gaaclarke) [reply shell/platform/android/io/flutter/embedding/android/FlutterFragment.java]: done

File: shell/platform/android/io/flutter/embedding/android/FlutterActivity.java (hunk 2 above)
Comment (gaaclarke) [reply shell/platform/android/io/flutter/embedding/android/FlutterActivity.java]: done

**Comment by gaaclarke** [comment]
> I guess that we also need to add `getDartEntrypointLibraryUri ` to `FlutterFragmentActivity`, just like `getDartEntrypointFunctionName`.

It should already exist by virtue of FlutterFragmentActivity inheriting from FlutterActivity.  It looks like FlutterFragmentActivity's getDartEntrypointFunctionName is the same as FlutterActivity's.  I'm not sure why it overloads it with the exact same code.  Probably a quirk of changing inheritance.

**Comment by ColdPaleLight** [comment]
> 

Sorry, I didn't express clearly. Since the superclass of `FlutterFragmentActivity` is `FragmentActivity` instead of `FlutterActivity`, I think we still need to add this feature to `FlutterFragmentActivity` :-)  
Here is a PR I filed to do this https://github.com/flutter/engine/pull/30790

**Pull request merged**